build_tools/docker/cmake-emscripten/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# An image for building IREE through Emscripten using CMake.
+
+FROM gcr.io/iree-oss/cmake@sha256:9d9953acf5ca0cf1ff3e8de32f10f24dfab1c4e8ec5d1fc047f556024ee4bed6 AS final
+
+# See also
+#   * https://github.com/emscripten-core/emsdk/blob/main/docker/Dockerfile
+#   * https://hub.docker.com/r/emscripten/emsdk
+
+ARG EMSDK_COMMIT=5c0e31a03a72136ccaa761c408baf5a640f942ec
+ARG SDK_VERSION=2.0.25
+
+# Follow https://emscripten.org/docs/getting_started/downloads.html.
+RUN git clone https://github.com/emscripten-core/emsdk
+RUN cd emsdk && git checkout "${EMSDK_COMMIT?}" && \
+    ./emsdk install ${SDK_VERSION?} && \
+    ./emsdk activate ${SDK_VERSION?}
```

Comment:
What files/directories does this actually create? Is it feasible to do this in an earlier build stage and then copy it into the final build? Especially not thrilled about having the entire git history in our image. If the above isn't feasible, can we delete the git repository after?

Reply:
Here are the contents of the `emsdk` folder on Windows (using [WinDirStat](https://windirstat.net/)):

![image](https://user-images.githubusercontent.com/4010439/127187558-721649c3-04e7-4de7-b2a9-92b6f7329be9.png)

The .git folder is tiny compared to the binaries (clang, wasm-ld, etc.) and node_modules folders.

We could try pruning it or deriving from https://hub.docker.com/r/emscripten/emsdk, but I don't think we'd be able to save much space or configuration complexity either way.

Reply:
Oh boy that's big. Would be nice if we could only grab the binaries we needed, but ah well :-/ Still smaller than the TF image